lib/screens/ol_crowding/logger.ex

```diff
@@ -0,0 +1,67 @@
+defmodule Screens.OlCrowding.Logger do
+  @moduledoc false
+
+  require Logger
+  use GenServer
+
+  alias Screens.Predictions.Prediction
+
+  def start_link(opts) do
+    GenServer.start_link(__MODULE__, opts)
+  end
+
+  @impl true
+  def init(state) do
+    schedule_run()
+
+    {:ok, state}
+  end
+
+  @impl true
+  def handle_info(
+        :run,
+        %{
+          original_crowding_levels: original_crowding_levels,
+          prediction: prediction,
+          logging_options: %{
+            is_real_screen: true,
+            screen_id: screen_id,
+            triptych_pane: triptych_pane
+          },
+          train_crowding_config: train_crowding_config,
+          fetch_predictions_fn: fetch_predictions_fn,
+          fetch_parent_stop_id_fn: fetch_parent_stop_id_fn,
+          fetch_params: fetch_params
+        } = state
+      ) do
+    schedule_run()
+
+    {:ok, predictions} = fetch_predictions_fn.(fetch_params)
+    next_train_prediction = List.first(predictions)
+    crowding_levels = Enum.map_join(prediction.vehicle.carriages, ",", & &1.occupancy_status)
```

Comment:
I might be off-base here but, since we fetch a new set of predictions in this function and don't just go off of the predictions that originally triggered this genserver to be spawned: is there a chance that `vehicle.carriages` is `nil` here? Like some weird thing happened where they lost crowding info in the time between the original predictions fetch and this one.

Followup Q: do we even care if that happens? The genserver would crash, but it's already an exceptional case, so maybe that's ok. "let it fail", right?

Reply:
So I do know that the default for `carriages` is an [empty list](https://github.com/mbta/api/blob/df7e38a54ebb9deb2c68d47af3f48de2e0c26dc9/apps/parse/lib/parse/vehicle_positions.ex#L70). So we won't have to ever worry about it being `nil`. I think this is good, too, because then we will log when this happens because the `original_crowding_levels` won't match the new `crowding_levels`.